Sources/SwifterSwift/SwiftStdlib/StringExtensions.swift

```diff
@@ -586,6 +586,36 @@ public extension String {
     }
     #endif
 
+    #if canImport(Foundation)
+    // https://www.hackingwithswift.com/articles/141/8-useful-swift-extensions
+    /// SwifterSwift: Replaces a set number of occurrences of a string with replacement text.
+    ///
+    ///        "I swift swift".replacingOccurrences(of: "swift", with: "love", count: 1) -> "I love swift."
+    /// - Parameters:
+    ///     - search: the string to be replaced
+    ///     - replacement: the replacement string
+    ///     - maxReplacements: the maximum number of times to replace the string, which must be greater than 0
+    /// - Returns: The new string with the replacement text swapped the specified amount of times.
+    func replacingOccurrences(of search: String, with replacement: String, count maxReplacements: Int) -> String {
+        guard maxReplacements > 0 else { return self }
+
+        var count = 0
+        var returnValue = self
+
+        while let range = returnValue.range(of: search) {
+            returnValue = returnValue.replacingCharacters(in: range, with: replacement)
```

Comment:
Also, we are copying the string multiple types here, maybe there is a in-place mutating version of `replacingCharacters` ?

Reply:
I could not find an in-place mutating version in the standard library or Foundation. Do you have any other suggestions?

Reply:
What about [replaceSubrange](https://developer.apple.com/documentation/swift/string/1641462-replacesubrange)?